enumfields/fields.py

```diff
@@ -28,10 +31,47 @@ def __get__(self, obj, type=None):
         return obj.__dict__[self.field.name]
 
     def __set__(self, obj, value):
-        obj.__dict__[self.field.name] = self.field.to_python(value)
-
-
-class EnumFieldMixin:
+        initial_field_name = INITIAL_FIELD_TMP.format(self.field.name)
+        python_value = self.field.to_python(value)
+        obj.__dict__.setdefault(initial_field_name, python_value)
+        obj.__dict__[self.field.name] = python_value
+
+
+class EnumFieldValidationMixin:
+
+    def _validate_next_value(self, value, model_instance):
+        initial_field_name = INITIAL_FIELD_TMP.format(self.name)
+        previous_choice = getattr(model_instance, initial_field_name)
+        if (
+            previous_choice is not None
+            and value != previous_choice
+            and previous_choice.next is not None
```

Comment:
there is difference between empty set and None. Right?

Reply:
yes. `next=None` is by default -> every state is allowed. `next=set()` -> from this state there's no transition to other state